src/Bicep.LangServer.IntegrationTests/CompletionTests.cs

```diff
@@ -51,18 +52,94 @@ public async Task EmptyFileShouldProduceDeclarationCompletions()
 
             var actual = await GetActualCompletions(client, uri, new Position(0, 0));
             var actualLocation = FileHelper.SaveResultFile(this.TestContext, $"{this.TestContext.TestName}_{expectedSetName}", actual.ToString(Formatting.Indented));
-            
+
             var expectedStr = DataSets.Completions.TryGetValue(GetFullSetName(expectedSetName));
             if (expectedStr == null)
             {
                 throw new AssertFailedException($"The completion set '{expectedSetName}' does not exist.");
             }
-            
+
             var expected = JToken.Parse(expectedStr);
 
             actual.Should().EqualWithJsonDiffOutput(TestContext, expected, GetGlobalCompletionSetPath(expectedSetName), actualLocation);
         }
 
+        [TestMethod]
```

Comment:
Is there a way we could make this into a `DataTestMethod`, so that each snippet is tested independently?

Rather than using `GetDeclationSnippetCompletionsAsync()` as the source of completions, I imagine you would need to start with the SnippetTemplate files as the data source.

Reply:
Hmmm I intentionally did this to avoid the scenario where someone actually checks in a snippet without adding an entry to the test assembly. Am I missing something?

Reply:
For user-submitted examples we:
1. Search the directory by file extension so that we don't miss anything.
2. Verify that every `.bicep` example has a corresponding `.json` file.

Would that work here? e.g. you find all the valid snippet files, make sure each one has a corresponding snippet template, and then run the test.

Reply:
Yeah +1 on this. The data rows for the data-driven tests can be generated dynamically via code, so it should be doable. One neat thing about it is that it becomes much easier to see if multiple snippets failed because each one appears as a different test then. (You can even control the display name of each row.)

---
In reply to: [600902530](https://github.com/Azure/bicep/pull/1999#discussion_r600902530) [](ancestors = 600902530)